### DISCUSSION TITLE
Wiki.js doesn't support custom git ssh port?

### DISCUSSION THREAD

**Discussion by lukonin** [requarks/wiki#3495] [Error / Bug Report]
I'm setting up a Repository URI as: git@domain:8722/username/repo.git
But in the tshark dump I see:

172.18.0.3 → 192.168.25.105 TCP 68 40250 → 22

172.18.0.3 is IP of Wiki.js in Docker.
192.168.25.105 is IP of Docker server.

**Steps to reproduce the behavior:**
1. Go to 'Settings' → 'Storage' → 'Git'
2. Set the 'Repository URI' with a custom SSH port.
3. Run tshark on Docker server.
4. See a default SSH port instead of given in 'Repository URI'.

**Host Info:**
 - OS: Debian 10.6, Docker
 - Wiki.js version: 2.5.159
 - Database engine: postgres 11.9

**Comment by derdeagle** [comment]
I have the same problem and with some manual adjustments it is working now. **Explicitly not a fix** but maybe interesting to get rid of a blocker.

1. Login as the node-user `docker exec -it <container> bash`
2. Switch to the Git directory (for me it was `cd /wiki/data/repo`)
3. Create a file there and push it to the repo (to initialize everything)
3.1 `echo "testfile" > test`
3.2 `git add test`
3.3 `git commit -m "init"`
3.4 `git push origin <branch>` (the branch in my case was **master**)
3.5 `git rm test`
3.6 `git commit -m "cleanup init"`
3.7 `git push origin <branch>`

After that it was working for me.
What is now in the `.git/config` that wasn't there before the manual changes is

```ini
[branch "master"]
	remote = origin
	merge = refs/heads/master
```

**Comment by aislam** [comment]
As of 2.5.295 Wikijs does not support non standard SSH port for git. I also ignores .ssh/config port configuration. I believe it is a problem coming from simple-git node module. Non standard SSH port is quite common and considered a good security practice. Hopefully, it will be fixed soon. I get an error like the following:

```
ssh: connect to host [hostname] port 22: Connection refused fatal: Could not read from remote repository. Please make sure you have the correct access rights and the repository exists. error: Could not fetch origin
```
Keep up the awesome work!

**Comment by curana** [comment]
It would be great if this could be solved. Especially in a LAN it is not uncommon to use custom ports.

**Reply by BPowell76** [reply]
I could see using an atypical port if the remote repo was on some public server, but if it's on a LAN what does it accomplish?

**Reply by curana** [reply]
The standard SSH port is already used on the homeserver. So it would be convenient.

**Reply by BPowell76** [reply]
Does that cause issues? I'm asking because I don't know. I know with a web server like Apache I can serve multiple sites from ports 80 and 443 with no issues. Is SSH not that "sophisticated"?

**Reply by curana** [reply]
SSH is IP-bound, not domain-bound.
